File: solutions/Java/number_of_1_bits_loop.java
public class NumberOf1BitsLoop {
    public static int numberOf1BitsLoop(int n) {
        int count = 0;
        while (n != 0) {
            count += n & 1;
            n >>>= 1;
        }
        return count;
    }

    public static void main(String[] args) {
        System.out.println(numberOf1BitsLoop(11));  // 3
    }
}

File: solutions/Python/number_of_1_bits_loop.py
def number_of_1_bits_loop(n: int) -> int:
    """Count 1 bits by checking each bit."""
    count = 0
    while n:
        count += n & 1
        n >>= 1
    return count

# Test cases
print(number_of_1_bits_loop(11))  # 3
print(number_of_1_bits_loop(128))  # 1
